src/geom/shape/triangle.rs

```diff
@@ -116,7 +116,7 @@ impl Triangle {
         }
 
         let q = rel_pos.cross(&e1);
-        let v = inv_e1_dot_d_cross_e2 * ray.dir().dot(&q.dir());
+        let v = inv_e1_dot_d_cross_e2 * ray.dir().dot_vec(&q);
```

Comment:
I won't ask this any more times, but again is `dot_vec` more optimal than `dot` etc.?

Reply:
Is not, probably less optimal, but I thought it was solving a problem, as cannot do polymorphism by accepting general types.